### PR TITLE
Discovery service based on proto endpoint

### LINKED ISSUE #13: Discovery service implementation
We need to create a service to get to know which network structure is available to the client to get information from. This would eventually be an endpoint where we connect to get a tree of all the possibles microservices and nodes to show, and then select a specific one to start consuming metrics.

### PR DESCRIPTION
**Pull request by calvellido**
* Add fetch-proto script
   * Add proto files fetching node script
   * Update README with specifics on how to fetch proto files
   * Add axios dependency to be able make HTTP requests
   * Add metrics proto file to proto directory

* Use google-protobuf JS runtime
  * Update gen-proto-code script to use protoc binary.
  * Update npm scripts.
  * Update README.
  * Update dependencies

* Fix error message on gen-proto-code script 

* Add JS generated classes from proto descriptors

* Add microservices discovery service
  * Add a microservices information fetching method, deserializing a meesaged based on the proto associated with that endpoint.
  * Consume that service on the app main component.
  * Set an environment variable that points to that microservices and nodes discovery information.

This closes #13

### CODE REVIEW

File: build-scripts/fetch-proto.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      description: '<proto-file-source-url>',
+      requiresArg: true,
+      default: originProtoPath,
+      string: true,
+      coerce: arg => {
+        try {
+          const urlObject = new URL(arg);
+          return urlObject;
+        } catch (e) {
+          // throw new Error(`Error: ${arg} is not a valid URL path`);
```
Comment (anamariamv) [inline build-scripts/fetch-proto.js]: Remove comment

File: build-scripts/fetch-proto.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    .then((response) => {
+      const protoFilename = `${originProtoPath.pathname.match(/([^\/]*)\/*$/)[1]}.proto`;
+      fs.writeFileSync(path.join(destProtoPath, `${protoFilename}`), response.data);
+      const successMessage = `${protoFilename} fetched`;
+      spinner.succeed(successMessage);
+      return successMessage;
+    })
+    .catch((error) => {
+      spinner.fail(`Error connecting to: ${error.config.url}`);
+      // shell.echo(`Error connecting to: ${error.config.url}`);
```
Comment (anamariamv) [inline build-scripts/fetch-proto.js]: same above

File: src/environments/environment.prod.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export const environment = {
   production: true,
   // metricsEndpoint: 'ws://localhost:8080/metrics'
-  metricsEndpoint: 'ws://localhost:3000'
+  metricsEndpoint: 'ws://localhost:3000',
```
Comment (anamariamv) [inline src/environments/environment.prod.ts]: Comment

File: src/environments/environment.ts [hunk 4]
```diff
@@ -6,5 +6,6 @@
 export const environment = {
   production: false,
   // metricsEndpoint: 'ws://localhost:8080/metrics'
-  metricsEndpoint: 'ws://localhost:3000'
+  metricsEndpoint: 'ws://localhost:3000',
```
Comment (anamariamv) [inline src/environments/environment.ts]: Same above

**Review by anamariamv — APPROVED** [review]
Minor comments.

File: build-scripts/fetch-proto.js (hunk 1 above)
Comment (calvellido) [reply build-scripts/fetch-proto.js]: Adressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/30/commits/59ef685be154fab90594badf674b77afc29082d8

File: build-scripts/fetch-proto.js (hunk 2 above)
Comment (calvellido) [reply build-scripts/fetch-proto.js]: Adressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/30/commits/59ef685be154fab90594badf674b77afc29082d8

File: src/environments/environment.prod.ts (hunk 3 above)
Comment (calvellido) [reply src/environments/environment.prod.ts]: Adressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/30/commits/59ef685be154fab90594badf674b77afc29082d8

File: src/environments/environment.ts (hunk 4 above)
Comment (calvellido) [reply src/environments/environment.ts]: Adressed on https://github.com/frees-io/freestyle-opscenter-webclient/pull/30/commits/59ef685be154fab90594badf674b77afc29082d8

**Pull request merged**